server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -81,6 +87,10 @@ final class FieldTypeLookup {
             // this will override concrete fields with runtime fields that have the same name
             fullNameToFieldType.put(fieldType.name(), fieldType);
         }
+        this.fullNameToFieldType = Map.copyOf(fullNameToFieldType);
```

Comment:
Can you add a comment here explaining that using `Map.copyOf` results in smaller memory usage and faster access?